rust_src/src/lisp.rs

```diff
@@ -836,6 +848,15 @@ impl LispObject {
         )
     }
 
+    /// Check if Lisp object is a character or not.
+    /// Similar to CHECK_CHARACTER
+    #[inline]
+    pub fn is_character_or_error(self) -> () {
```

Comment:
Please make this `as_character_or_error`. Then you can make the cast to `fixnum` and ultimately to `Codepoint` here already.